modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -431,9 +433,11 @@ const FrontendDataSet = ({
 				}
 			});
 
+			onSelect(newSelectedItems);
+
```

Comment:
What will happen in List view where we [already implement onSelect](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/list/List.js#L96)? Won't it trigger twice?

Reply:
Hey @markocikos I remembered why I had chosen to introduce a new prop, as I had checked other `onSelect` deep down back in the days and so we were using it with single selected items one by one. In our case what we needed is the whole array of `selectedItems` as soon as it was freshly updated, and since I didn't want to break other implementations I resorted to "invent" a new prop just for that purpose.

I rolled back to `onItemSelected` and simply placed it in the already present `useEffect` for `selectedItems` updates, if that's ok for you.

Reply:
Why can't new `onSelect` just return an array with one item? It's a new API, so we won't break anything, right?

Reply:
Because, for instance, for `Table` I'd need to propagate and call `onSelect` across all of the points where it needs to trigger with the correct logic (checkbox/radio for rows, checkbox in `Head` to select/deselect all for `multiple`, `Select All` link in management bar). And what's worse, since `onSelect` only returns 1 single item, in some outer `propsTransformer` users would also have to "keep the count" of what to add and what to remove just based on the latest `item` selected as they wouldn't know if it's been checked or unchecked (again for multiple, for `Head` multi-selection or `Select All` in the management bar it would be even harder unless breaking the current `onSelect` API). 

To me all of this looks like a complication, as a user may simply need to access the **state of the art** `selectedItems` as a whole, as the `FrontendDataSet` is already doing all of the math to add/remove according to the different views/components/partials logics.

Given all of this, this is why I proposed to trigger some user-managed prop to trigger in the specific effect that's keeping track of `selectedItems` state updates.

Now, since `onSelect` indeed is used and its API is that of returning 1 single item, if I use it it'll break many other tables across Portal (tested). So here's why I simply proposed a brand-new `onItemSelected`.

If you prefer we can also take this on a quick call, just in case I'm missing anything.

Thanks again!